cd /storage/home/users/pjt6/newton/comparative_genomics/all_nems

#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa_hgly.fa.RBBH.tab globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa hgly.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa_Gpal_newton_newton.proteins.fa.RBBH.tab globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa Gpal_newton_newton.proteins.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa_nGr.v1.1.augustus.manual.2015_04_02.aa.fa.RBBH.tab globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa nGr.v1.1.augustus.manual.2015_04_02.aa.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa_Gpal.v1.0.AA.fa.RBBH.tab globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa Gpal.v1.0.AA.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o hgly.fa_globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa.RBBH.tab hgly.fa globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o hgly.fa_Gpal_newton_newton.proteins.fa.RBBH.tab hgly.fa Gpal_newton_newton.proteins.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o hgly.fa_nGr.v1.1.augustus.manual.2015_04_02.aa.fa.RBBH.tab hgly.fa nGr.v1.1.augustus.manual.2015_04_02.aa.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o hgly.fa_Gpal.v1.0.AA.fa.RBBH.tab hgly.fa Gpal.v1.0.AA.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o Gpal_newton_newton.proteins.fa_globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa.RBBH.tab Gpal_newton_newton.proteins.fa globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o Gpal_newton_newton.proteins.fa_hgly.fa.RBBH.tab Gpal_newton_newton.proteins.fa hgly.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o Gpal_newton_newton.proteins.fa_nGr.v1.1.augustus.manual.2015_04_02.aa.fa.RBBH.tab Gpal_newton_newton.proteins.fa nGr.v1.1.augustus.manual.2015_04_02.aa.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o Gpal_newton_newton.proteins.fa_Gpal.v1.0.AA.fa.RBBH.tab Gpal_newton_newton.proteins.fa Gpal.v1.0.AA.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o nGr.v1.1.augustus.manual.2015_04_02.aa.fa_globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa.RBBH.tab nGr.v1.1.augustus.manual.2015_04_02.aa.fa globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o nGr.v1.1.augustus.manual.2015_04_02.aa.fa_hgly.fa.RBBH.tab nGr.v1.1.augustus.manual.2015_04_02.aa.fa hgly.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o nGr.v1.1.augustus.manual.2015_04_02.aa.fa_Gpal_newton_newton.proteins.fa.RBBH.tab nGr.v1.1.augustus.manual.2015_04_02.aa.fa Gpal_newton_newton.proteins.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o nGr.v1.1.augustus.manual.2015_04_02.aa.fa_Gpal.v1.0.AA.fa.RBBH.tab nGr.v1.1.augustus.manual.2015_04_02.aa.fa Gpal.v1.0.AA.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o Gpal.v1.0.AA.fa_globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa.RBBH.tab Gpal.v1.0.AA.fa globodera_rostochiensis.PRJEB13504.WBPS13.protein.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o Gpal.v1.0.AA.fa_hgly.fa.RBBH.tab Gpal.v1.0.AA.fa hgly.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o Gpal.v1.0.AA.fa_Gpal_newton_newton.proteins.fa.RBBH.tab Gpal.v1.0.AA.fa Gpal_newton_newton.proteins.fa 
#python /storage/home/users/pjt6/misc_python/BLAST_output_parsing/Blast_RBH_two_fasta_file_evalue.py  --threads 16  -o Gpal.v1.0.AA.fa_nGr.v1.1.augustus.manual.2015_04_02.aa.fa.RBBH.tab Gpal.v1.0.AA.fa nGr.v1.1.augustus.manual.2015_04_02.aa.fa 


cd /storage/home/users/pjt6/newton/comparative_genomics/all_nems/RBBH_clustering

conda activate bio_ortho
cat ../*.tab | cut -f1,2,11 > all_vs_all.abc


values="10 11 12 13 14 15 16 17 18 19 20"
for v in ${values}
do
	echo "2) filtering BLAST at ${v}"
	# run the five prime fixer tool
	filter_blast="python /storage/home/users/pjt6/misc_python/MCL_clusters_get_seq/Remove_low_quality_blast_matches.py 
				  all_vs_all.abc all_results_eval_1e-${v}.abc 
				  1e-${v}" 
	echo ${filter_blast}
	eval ${filter_blast}
	wait
done

# deliberately putting this is a separted loop!!!!
for v in ${values}
do
	echo "3) mcxload"
	# run the five prime fixer tool
	mcxload_cmd="mcxload 
	-abc all_results_eval_1e-${v}.abc 
	--stream-mirror 
	--stream-neg-log10 
	-stream-tf 'ceil(200)' 
	-o all_results_eval_1e-${v}.mci 
	-write-tab 
	all_results_eval_1e-${v}.abc.tab" 
	echo ${mcxload_cmd}
	eval ${mcxload_cmd}
	wait
done


# deliberately putting this is a separted loop!!!!
for v in ${values}
do
	echo "4) mcxload"
	# run the five prime fixer tool
	mcl_cmd="mcl 
	all_results_eval_1e-${v}.mci 
	-I 6 
	-use-tab all_results_eval_1e-${v}.abc.tab 
	-o all_results_eval_1e-${v}.abc_seq.mci
" 
	echo ${mcl_cmd}
	eval ${mcl_cmd}
	wait
done

# deliberately putting this is a separted loop!!!!
for v in ${values}
do
	echo "5) running MCL"
	# run the five prime fixer tool
	mcl_cmd="mcl 
	all_results_eval_1e-${v}.mci 
	-I 6 
	-use-tab all_results_eval_1e-${v}.abc.tab 
	-o all_results_eval_1e-${v}.abc_seq.mci
" 
	echo ${mcl_cmd}
	eval ${mcl_cmd}
	wait
done



# deliberately putting this is a separted loop!!!!
for v in ${values}
do
	echo "6) writing graph"
	# run the five prime fixer tool
	mcl_cmd="mcl all_results_eval_1e-${v}.mci 
	--d 
	-write-graphx 
	all_results_eval_1e-${v}.mci_Outputgraph.txt 
	-I 6" 
	echo ${mcl_cmd}
	eval ${mcl_cmd}
	wait
done


# deliberately putting this is a separted loop!!!!
for v in ${values}
do
	echo "7) mcl dump"
	# run the five prime fixer tool
	mcl_cmd="mcxdump 
	-icl out.all_results_eval_1e-${v}.mci.I60 
	-tabr all_results_eval_1e-${v}.abc.tab 
	-o all_results_eval_1e-${v}.mci.I60" 
	echo ${mcl_cmd}
	eval ${mcl_cmd}
	wait
done

# deliberately putting this is a separted loop!!!!
for v in ${values}
do
	echo "7) mcl dump"
	# run the five prime fixer tool
	cluster_species_count_cmd='python mcl_to_cafe.py 
	-i all_results_eval_1e-${v}.abc_seq.mci
	-sp "APISUM_N116,APISUM_PS01,Api"
	-o all_results_eval_1e-${v}.SPECIES'
	echo ${mcl_cmd}
	#eval ${mcl_cmd}'
	wait
done
mkdir gros

/shelf/apps/pjt6/conda/envs/python27/bin/python get_clustr_sequences.py 


###########################################################################################

cd /storage/home/users/pjt6/newton/comparative_genomics/all_nems/RBBH_clustering/effectors
rm *fasta_*
rm -rf back_translated

filenames=*.fasta



for f in ${filenames}
do
	#echo "Running muscle ${f}"
	cmd="muscle -in ${f} -out ${f}_aligned.fasta" 
	echo ${cmd}
	eval ${cmd}
	wait
	
done
###############################
###############################
# purge bad aligning sequences: https://github.com/dukecomeback/bad-sequence-remover


filenames2=*aligned.fasta
for file in ${filenames2}
do
	echo "Running seq purging ${f}"
	cmd="perl /storage/home/users/pjt6/bad-sequence-remover/badseq_remover.pl
    -in ${file} 
    -out ${file}_purges.fasta
    -exp GPALN" 
	echo ${cmd}
	eval ${cmd}
	wait
done

###############

filenames2=*_purges.fasta
for file in ${filenames2}
do
	echo "Running muscle ${f}"
	cmd="muscle -in ${file} -out ${file}_refine.fasta -refine" 
	echo ${cmd}
	eval ${cmd}
	wait
done
mkdir back_translated

python /storage/home/users/pjt6/newton/comparative_genomics/all_nems/Align_back_translate.py

#####################
cd back_translated/
#filenames3=*.fasta
#for file in ${filenames3}
#do
#	echo "Running trimal ${file}"
#	cmd="/storage/home/users/pjt6/trimAl/source/trimal  
#    -in ${file}  
#    -out ${file}.phy
#    -phylip 
#    -gappyout" 
#	echo ${cmd}
#	eval ${cmd}
#	wait
#done

cd back_translated/
filenames3=*translated.fasta
for file in ${filenames3}
do
	echo "Running trimal ${file}"
	cmd="/storage/home/users/pjt6/trimAl/source/trimal  
    -in ${file}  
    -out ${file}.phy
    -phylip 
    -nogaps" 
	echo ${cmd}
	eval ${cmd}
	wait
done

mkdir phy_files

mv *.phy ./phy_files/

cd phy_files


filenames=*.phy

# phyml
export PATH=//storage/home/users/pjt6/phyml-20140723/src/:$PATH

#codonphyml
export PATH=/storage/home/users/pjt6/codonPHYML_dev/src/:$PATH



## nucleotide version
mkdir files_done
for f in ${filenames}
do
	echo "Running codonphyml ${f}"
	cmd="codonphyml -i ${f} -m GY --fmodel F3X4 -t e -f empirical -w g -a e" #--optBrent 3
	echo ${cmd}
	eval ${cmd}
	wait
	cmd2="mv ${f} ./files_done"
	echo ${cmd2}
	eval ${cmd2}
done

grep "Probability: o3=" -H *_stats.txt > ~/newton/comparative_genomics/all_nems/effector_no_gaps_RBBH_orthofinder_DN_DS.txt



cd /storage/home/users/pjt6/newton/comparative_genomics/all_nems/RBBH_clustering/busco/
rm *fasta_*
rm -rf back_translated
filenames=*.fasta



for f in ${filenames}
do
	#echo "Running muscle ${f}"
	cmd="muscle -in ${f} -out ${f}_aligned.fasta" 
	echo ${cmd}
	eval ${cmd}
	wait
	
done
###############################
###############################
# purge bad aligning sequences: https://github.com/dukecomeback/bad-sequence-remover


filenames2=*aligned.fasta
for file in ${filenames2}
do
	echo "Running seq purging ${f}"
	cmd="perl /storage/home/users/pjt6/bad-sequence-remover/badseq_remover.pl
    -in ${file} 
    -out ${file}_purges.fasta
    -exp GPALN" 
	echo ${cmd}
	eval ${cmd}
	wait
done

###############

filenames2=*_purges.fasta
for file in ${filenames2}
do
	echo "Running muscle ${f}"
	cmd="muscle -in ${file} -out ${file}_refine.fasta -refine" 
	echo ${cmd}
	eval ${cmd}
	wait
done
mkdir back_translated

python /storage/home/users/pjt6/newton/comparative_genomics/all_nems/Align_back_translate.py

#####################
cd back_translated/
#filenames3=*.fasta
#for file in ${filenames3}
#do
#	echo "Running trimal ${file}"
#	cmd="/storage/home/users/pjt6/trimAl/source/trimal  
#    -in ${file}  
#    -out ${file}.phy
#    -phylip 
#    -gappyout" 
#	echo ${cmd}
#	eval ${cmd}
#	wait
#done

cd back_translated/
filenames3=*translated.fasta
for file in ${filenames3}
do
	echo "Running trimal ${file}"
	cmd="/storage/home/users/pjt6/trimAl/source/trimal  
    -in ${file}  
    -out ${file}.phy
    -phylip 
    -nogaps" 
	echo ${cmd}
	eval ${cmd}
	wait
done

mkdir phy_files

mv *.phy ./phy_files/

cd phy_files


filenames=*.phy

# phyml
export PATH=//storage/home/users/pjt6/phyml-20140723/src/:$PATH

#codonphyml
export PATH=/storage/home/users/pjt6/codonPHYML_dev/src/:$PATH



## nucleotide version
mkdir files_done
for f in ${filenames}
do
	echo "Running codonphyml ${f}"
	cmd="codonphyml -i ${f} -m GY --fmodel F3X4 -t e -f empirical -w g -a e" #--optBrent 3
	echo ${cmd}
	eval ${cmd}
	wait
	cmd2="mv ${f} ./files_done"
	echo ${cmd2}
	eval ${cmd2}
done

grep "Probability: o3=" -H *_stats.txt > ~/newton/comparative_genomics/all_nems/busco_no_gaps_RBBH_orthofinder_DN_DS.txt



cd /storage/home/users/pjt6/newton/comparative_genomics/all_nems/RBBH_clustering/sub
rm *fasta_*
rm -rf back_translated

filenames=*.fasta



for f in ${filenames}
do
	#echo "Running muscle ${f}"
	cmd="muscle -in ${f} -out ${f}_aligned.fasta" 
	echo ${cmd}
	eval ${cmd}
	wait
	
done
###############################
###############################
# purge bad aligning sequences: https://github.com/dukecomeback/bad-sequence-remover


filenames2=*aligned.fasta
for file in ${filenames2}
do
	echo "Running seq purging ${f}"
	cmd="perl /storage/home/users/pjt6/bad-sequence-remover/badseq_remover.pl
    -in ${file} 
    -out ${file}_purges.fasta
    -exp GPALN" 
	echo ${cmd}
	eval ${cmd}
	wait
done

###############

filenames2=*_purges.fasta
for file in ${filenames2}
do
	echo "Running muscle ${f}"
	cmd="muscle -in ${file} -out ${file}_refine.fasta -refine" 
	echo ${cmd}
	eval ${cmd}
	wait
done
mkdir back_translated

python /storage/home/users/pjt6/newton/comparative_genomics/all_nems/Align_back_translate.py

#####################
cd back_translated/
#filenames3=*.fasta
#for file in ${filenames3}
#do
#	echo "Running trimal ${file}"
#	cmd="/storage/home/users/pjt6/trimAl/source/trimal  
#    -in ${file}  
#    -out ${file}.phy
#    -phylip 
#    -gappyout" 
#	echo ${cmd}
#	eval ${cmd}
#	wait
#done

cd back_translated/
filenames3=*translated.fasta
for file in ${filenames3}
do
	echo "Running trimal ${file}"
	cmd="/storage/home/users/pjt6/trimAl/source/trimal  
    -in ${file}  
    -out ${file}.phy
    -phylip 
    -nogaps" 
	echo ${cmd}
	eval ${cmd}
	wait
done

mkdir phy_files

mv *.phy ./phy_files/

cd phy_files


filenames=*.phy

# phyml
export PATH=//storage/home/users/pjt6/phyml-20140723/src/:$PATH

#codonphyml
export PATH=/storage/home/users/pjt6/codonPHYML_dev/src/:$PATH



## nucleotide version
mkdir files_done
for f in ${filenames}
do
	echo "Running codonphyml ${f}"
	cmd="codonphyml -i ${f} -m GY --fmodel F3X4 -t e -f empirical -w g -a e" #--optBrent 3
	echo ${cmd}
	eval ${cmd}
	wait
	cmd2="mv ${f} ./files_done"
	echo ${cmd2}
	eval ${cmd2}
done

grep "Probability: o3=" -H *_stats.txt > ~/newton/comparative_genomics/all_nems/sub_no_gaps_RBBH_orthofinder_DN_DS.txt



cd /storage/home/users/pjt6/newton/comparative_genomics/all_nems/RBBH_clustering/gros/
rm *fasta_*
rm -rf back_translated
filenames=*.fasta



for f in ${filenames}
do
	#echo "Running muscle ${f}"
	cmd="muscle -in ${f} -out ${f}_aligned.fasta" 
	echo ${cmd}
	eval ${cmd}
	wait
	
done
###############################
###############################
# purge bad aligning sequences: https://github.com/dukecomeback/bad-sequence-remover


filenames2=*aligned.fasta
for file in ${filenames2}
do
	echo "Running seq purging ${f}"
	cmd="perl /storage/home/users/pjt6/bad-sequence-remover/badseq_remover.pl
    -in ${file} 
    -out ${file}_purges.fasta
    -exp GPALN" 
	echo ${cmd}
	eval ${cmd}
	wait
done

###############

filenames2=*_purges.fasta
for file in ${filenames2}
do
	echo "Running muscle ${f}"
	cmd="muscle -in ${file} -out ${file}_refine.fasta -refine" 
	echo ${cmd}
	eval ${cmd}
	wait
done
mkdir back_translated

python /storage/home/users/pjt6/newton/comparative_genomics/all_nems/Align_back_translate.py

#####################
cd back_translated/
#filenames3=*.fasta
#for file in ${filenames3}
#do
#	echo "Running trimal ${file}"
#	cmd="/storage/home/users/pjt6/trimAl/source/trimal  
#    -in ${file}  
#    -out ${file}.phy
#    -phylip 
#    -gappyout" 
#	echo ${cmd}
#	eval ${cmd}
#	wait
#done

cd back_translated/
filenames3=*translated.fasta
for file in ${filenames3}
do
	echo "Running trimal ${file}"
	cmd="/storage/home/users/pjt6/trimAl/source/trimal  
    -in ${file}  
    -out ${file}.phy
    -phylip 
    -nogaps" 
	echo ${cmd}
	eval ${cmd}
	wait
done

mkdir phy_files

mv *.phy ./phy_files/

cd phy_files


filenames=*.phy

# phyml
export PATH=//storage/home/users/pjt6/phyml-20140723/src/:$PATH

#codonphyml
export PATH=/storage/home/users/pjt6/codonPHYML_dev/src/:$PATH



## nucleotide version
mkdir files_done
for f in ${filenames}
do
	echo "Running codonphyml ${f}"
	cmd="codonphyml -i ${f} -m GY --fmodel F3X4 -t e -f empirical -w g -a e" #--optBrent 3
	echo ${cmd}
	eval ${cmd}
	wait
	cmd2="mv ${f} ./files_done"
	echo ${cmd2}
	eval ${cmd2}
done

grep "Probability: o3=" -H *_stats.txt > ~/newton/comparative_genomics/all_nems/gros_no_gaps_RBBH_orthofinder_DN_DS.txt










